org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/TextEditUtils.java

```diff
@@ -206,4 +157,37 @@ private static <T> List<T> mergeSort(List<T> data, Comparator<T> comparator) {
 		return data;
 	}
 
+	/**
+	 * Returns the offset of the first whitespace that's found in the given range
+	 * [leftLimit,to] from the left of the to, and leftLimit otherwise.
+	 * 
+	 * When :
+	 * <ul>
+	 * <li>leftLimit is equals to -1 (ex : an element tag name which have not a
+	 * start tag bracket), the leftLimit is set 'to -1'.</li>
+	 * <li>to is equals to 0, the method returns -1</li>
+	 * </ul>
+	 * 
+	 * @param leftLimit the left limit range.
+	 * @param to        the to range.
+	 * 
+	 * @return the offset of the first whitespace that's found in the given range
+	 *         [leftLimit,to] from the left of the to, and leftLimit otherwise.
+	 */
+	public static int adjustOffsetWithLeftWhitespaces(int leftLimit, int to, String text) {
+		if (to == 0) {
+			return -1;
+		}
+		int limit = leftLimit != -1 ? leftLimit : 0;

```

Comment:
is this needed, and if so, what cases does it handle? I replaced it with `int limit = leftLimit`, and all the tests passed.

Reply:
Good catch, we can indeed simplify the code again. Now the method is very simply.  Thanks @datho7561 for your great suggestion.